fs/metrics/manager/manager.go

```diff
@@ -0,0 +1,44 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package manager
+
+import (
+	"time"
+
+	"github.com/awslabs/soci-snapshotter/fs/metrics/manager/monitors"
+)
+
+type FuseObservabilityManager struct {
```

Comment:
This structure encodes the metric dimensions into the type system and it seems to imply that we'll only ever want a single metric dimension. What stops us from having a single `Monitor` type that has a `Inc(metricName string, Labels ...LabelOpt)` where we can say `monitor.Inc(metricName, WithLayer(layerDigest))`? 

I think maybe the binary construction get's messed up here?

Reply:
Yeah, I was think about this for a while. I was initially working backwards from the metrics/dimensions that we have defined [here](https://github.com/awslabs/soci-snapshotter/blob/main/fs/metrics/common/metrics.go#L113-L168). 

Having a single monitor would mean that we would have to form an association between a set of labels and an underlying Prometheus metric. How would we know that some label corresponds to a single metric without explicitly storing dimension types into the monitor itself or maintaining some global map of labels->metric? For instance, `Inc(metricName, WithDigest(d))` can map to the layer counter **or** the image counter.